### ISSUE TITLE
Fix false positives for functions in font-weight-notation

### ISSUE THREAD

**Issue by iamnewton** [stylelint/stylelint#3855]
> What is the problem you're trying to solve?

I use a deep nested SCSS map with all of my theme options and have created a function to access things like font-weights.  I'm looking to include the `font-weight-notation` to be numeric unless its using a function I've whitelisted.

> What solution would you like to see?

As a developer, I would like to see the [font-weight-notation](https://stylelint.io/user-guide/rules/font-weight-notation/) expanded to include an option to `ignoreFunction` or maybe a regex since I use a deep nested SCSS map and function to access font-weights.

**Comment by jeddy3** [comment]
@iamnewton Thanks for the suggestion and for using the template

> I use a deep nested SCSS map with all of my theme options and have created a function to access things like font-weights

Can you provide some example code snippets of this? That'll help us work how best to solve this problem.

**Comment by iamnewton** [comment]
```
font-weight: map-deep-get($theme, typography, weight, semibold);
```

**Comment by jeddy3** [comment]
@iamnewton Thanks. I think an `ignoreValues: []` option is what we're after e.g.

```
{
  "rules": {
    "font-weight-notation": ["numeric", { ignoreValues: ["/^map-deep-get/"] }]
  }
}
```

Please consider contributing this option. There's [a section in the Developer Guide](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#adding-an-option-to-an-existing-rule) on how to get started. For reference, [PR 3748](https://github.com/stylelint/stylelint/pull/3748/files) adds a similar option to another rule.

**Comment by iamnewton** [comment]
You think `ingoreValues`?  Is `ignoreFunction` not the more consistent option?

**Comment by jeddy3** [comment]
An option focused on functions works too.

It'll be `ignoreFunctions: []` with an "s" to align with the conventions: https://stylelint.io/developer-guide/rules/#secondary

**Comment by jeddy3** [comment]
Upon revisiting this issue, I realise that it should have been classified as a bug.

This:

https://github.com/stylelint/stylelint/blob/b01ed25dfa3e8231a976eef76bd5e81fb535b1e9/lib/rules/font-weight-notation/index.js#L61-L67

And this:

https://github.com/stylelint/stylelint/blob/b01ed25dfa3e8231a976eef76bd5e81fb535b1e9/lib/rules/font-weight-notation/index.js#L108-L112

Are too loose.

We can fix the issue by:
- migrating to the value parser from `postcss.list.space` to find font-weight nodes
- returning early for `function` nodes except `var`, where we'll want to check the fallback

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.
